### ISSUE TITLE
Phpredis client with proxy sends extra value for sRandMember optional argument

### ISSUE THREAD

**Issue by Braxilior** [snc/SncRedisBundle#638]
Hi, I use the `snc/redis-bundle 3.5.2` with the phpredis client. I just installed the `symfony/proxy-manager-bridge` and some of my tests broke.
The problem comes from the handling of the optional argument of the `sRandMember` method.

Test code:
`$this->client->sRandMember('my_test_key');` 

With the proxy:
`snc_redis.DEBUG: Executing command "SRANDMEMBER my_test_key <null>"` 

Without the proxy:
`snc_redis.DEBUG: Executing command "SRANDMEMBER my_test_key"`

It looks similar to #620.

**Comment by ostrolucky** [comment]
3.x will no longer be maintained precisely because of issues like this. Help test is 4.x branch so that we can release it soon.

**Comment by Braxilior** [comment]
I tested the 4.x branch and I still had the same issue. The logs were the same than with the 3.5.2 version.

**Comment by ostrolucky** [comment]
Sorry, root cause would have to be analyzed by you, as I can't see correlation with proxies. Also, I don't really see an issue if <null> is included in message.

**Comment by Braxilior** [comment]
I think I understand the problem. The generated redis proxy uses `null` as the default count value:
```
public function sRandMember($key, $count = null)
{
    $this->initializerc4e66 && ($this->initializerc4e66->__invoke($valueHolder9e98e, $this, 'sRandMember', array('key' => $key, 'count' => $count), $this->initializerc4e66) || 1) && $this->valueHolder9e98e = $valueHolder9e98e;

    return $this->valueHolder9e98e->sRandMember($key, $count);
}
```


The expected behaviour described in the phpredis documentation:  
> If no count is provided, a random String value from the set will be returned

With the proxy, calling `sRandMember` without a count is the same thing than calling it with a count of 0 and the return value will always be an empty array.

The bug comes from a more general problem about php extensions and proxies: https://github.com/Ocramius/ProxyManager/issues/162#issuecomment-984591001
About `sRandMember`, this [commit](https://github.com/phpredis/phpredis/pull/1849/commits/69d3c86bf19b597a9af63ad7567d34dc970f3767) might fix the issue. It will be available in the 5.3.5 version of phpredis.
